docs/feature_combo.md

```diff
@@ -27,70 +42,255 @@ If you want to add a list, then you'd use something like this:
 ```c
 enum combos {
   AB_ESC,
-  JK_TAB
+  JK_TAB,
+  QW_SFT,
+  SD_LAYER,
 };
 
 const uint16_t PROGMEM ab_combo[] = {KC_A, KC_B, COMBO_END};
 const uint16_t PROGMEM jk_combo[] = {KC_J, KC_K, COMBO_END};
+const uint16_t PROGMEM qw_combo[] = {KC_Q, KC_W, COMBO_END};
+const uint16_t PROGMEM sd_combo[] = {KC_S, KC_D, COMBO_END};
 
 combo_t key_combos[COMBO_COUNT] = {
   [AB_ESC] = COMBO(ab_combo, KC_ESC),
-  [JK_TAB] = COMBO(jk_combo, KC_TAB)
+  [JK_TAB] = COMBO(jk_combo, KC_TAB),
+  [QW_SFT] = COMBO(qw_combo, KC_LSFT)
+  [SD_LAYER] = COMBO(layer_combo, MO(_LAYER)),
 };
 ```
 
 For a more complicated implementation, you can use the `process_combo_event` function to add custom handling.
+Additionally, this example shows how you can leave `COMBO_COUNT` undefined.
 
 ```c
 enum combo_events {
-  ZC_COPY,
-  XV_PASTE
+  TH_THE,
+  BSPC_LSFT_CLEAR,
+  COMBO_LENGTH
 };
+uint16_t COMBO_LEN = COMBO_LENGTH; // remove the COMBO_COUNT define and use this instead!
 
-const uint16_t PROGMEM copy_combo[] = {KC_Z, KC_C, COMBO_END};
-const uint16_t PROGMEM paste_combo[] = {KC_X, KC_V, COMBO_END};
+const uint16_t PROGMEM the_combo[] = {KC_BSPC, KC_T, COMBO_END};
+const uint16_t PROGMEM clear_line_combo[] = {KC_BSPC, KC_LSFT, COMBO_END};
 
-combo_t key_combos[COMBO_COUNT] = {
-  [ZC_COPY] = COMBO_ACTION(copy_combo),
-  [XV_PASTE] = COMBO_ACTION(paste_combo),
+combo_t key_combos[] = {
+  [TH_THE] = COMBO_ACTION(the_combo),
+  [BSPC_LSFT_CLEAR] = COMBO_ACTION(clear_line_combo),
 };
+/* COMBO_ACTION(x) is same as COMBO(x, KC_NO) */
 
 void process_combo_event(uint16_t combo_index, bool pressed) {
   switch(combo_index) {
-    case ZC_COPY:
+    case TH_THE:
       if (pressed) {
-        tap_code16(LCTL(KC_C));
+        SEND_STRING("the");
```

Comment:
Activating this combo while <kbd>Shift</kbd> (non-sticky/one-shot) is on would produce "THE" which is not what people would expect/desire. That said, I don't think the answer is to complicate the example code with modifier detection in order to solve the special case I've mentioned.

Instead, I suggest to replace `TH_THE` by `EM_EMAIL`; a combo triggered by <kbd>E</kbd> and <kbd>M</kbd> that calls `SEND_STRING("john.doe@example.com");`. 
It still showcases the use of `SEND_STRING` but removes the need for modifier handling because who needs to capitalize their email address?